[Security Solution][Entity Analytics][Risk Scoring] Add Risk Scoring Alert Filtering - Backend Implementation

abhishekbhatia1710 opened the pull request:
## 🎯 Summary

This PR implements the backend infrastructure for **Risk Scoring Alert Filtering**, enabling users to apply entity-specific KQL filters during risk score calculations. This enhancement allows for more targeted risk scoring by filtering alerts based on custom criteria.

## 🚀 What's New

### Core Features
- **Entity-Specific Filtering**: Apply KQL filters to specific entity types (host, user, service)
- **Backward Compatibility**: Existing configurations continue to work without changes
- **Graceful Error Handling**: Invalid KQL filters are silently ignored to prevent query failures
- **Migration Support**: Automatic migration of existing saved objects to include new filters field

### Technical Implementation

#### 1. **Saved Object Schema Enhancement**
- Added `filters` field to `risk-engine-configuration` saved object
- Implemented migration logic (version 3) for existing configurations
- Updated mappings version to 5

#### 2. **Enhanced Risk Scoring Logic**
- Created `buildFiltersForEntityType` helper function for entity-specific filter construction
- Integrated KQL parsing using `@kbn/es-query` utilities
- Applied filters at the aggregation level for optimal performance

#### 3. **API Endpoint Updates**
- Enhanced configuration endpoint to accept filters parameter
- Updated preview endpoint to support filter testing
- Maintained backward compatibility with existing API contracts

## 🏗️ Architecture

```mermaid
graph TD
    A[User Configuration] --> B[Risk Engine Saved Object]
    B --> C[buildFiltersForEntityType]
    C --> D[KQL Parser]
    D --> E[Elasticsearch Query]
    E --> F[Risk Score Aggregation]
    F --> G[Filtered Results]
    
    H[API Request] --> I[Route Handler]
    I --> J[Risk Score Service]
    J --> K[calculateRiskScores]
    K --> C
```

## 🔧 Filter Processing Flow

```mermaid
sequenceDiagram
    participant U as User
    participant API as API Endpoint
    participant S as Risk Score Service
    participant F as Filter Builder
    participant ES as Elasticsearch
    
    U->>API: POST /api/risk_scores/preview
    API->>S: calculateScores(filters)
    S->>F: buildFiltersForEntityType()
    F->>F: Parse KQL filters
    F->>F: Build ES queries
    F-->>S: Return filters array
    S->>ES: Execute aggregation with filters
    ES-->>S: Return filtered results
    S-->>API: Return risk scores
    API-->>U: Return response
```

## 🧪 Testing

### Unit Tests Added
- `buildFiltersForEntityType` function tests
- Saved object migration tests
- API endpoint filter parameter tests
- Error handling for invalid KQL filters

### Test Coverage
- ✅ Entity-specific filter application
- ✅ Multiple filters for same entity type
- ✅ Empty filter arrays handling
- ✅ Invalid KQL filter graceful handling
- ✅ Backward compatibility verification

## 📋 API Testing

### 1. **Configure Risk Engine with Filters**
```bash
curl -X PUT "http://localhost:5601/api/risk_score/engine/saved_object/configure" \
  -H "Content-Type: application/json" \
  -H "kbn-xsrf: true" \
  -H "Authorization : Basic ***" \
  -d '{
    "filters": [
      {
        "entity_types": ["host"],
        "filter": "agent.type: filebeat"
      },
      {
        "entity_types": ["user"],
        "filter": "user.name: ubuntu"
      }
    ]
  }'
```

**Expected Response:**
```json
{
  "risk_engine_saved_object_configured": true
}
```
**Get risk engine saved object configuration:**
```bash
curl --location --request GET 'http://localhost:5601/api/saved_objects/_find?type=risk-engine-configuration' \
--header 'elastic-api-version: 1' \
--header 'kbn-xsrf: true' \
--header 'x-elastic-internal-origin: true' \
--header 'Content-Type: application/json' \
--header 'Authorization: Basic ***' \
--data '{
  "range": {
    "start": "now-30d",
    "end": "now"
  },
  "includeClosedAlerts": false
}'
```
**Expected Response:**
```bash
{
    "page": 1,
    "per_page": 20,
    "total": 1,
    "saved_objects": [
        {
            "type": "risk-engine-configuration",
            "id": "36e6f63d-e5d0-4919-8d41-4988dd708754",
            "namespaces": [
                "default"
            ],
            "attributes": {
                "dataViewId": ".alerts-security.alerts-default",
                "enabled": false,
                "filter": {},
                "interval": "1h",
                "pageSize": 3500,
                "range": {
                    "start": "now-30d",
                    "end": "now"
                },
                "excludeAlertStatuses": [
                    "closed"
                ],
                "_meta": {
                    "mappingsVersion": 5
                },
                "filters": [
                    {
                        "entity_types": [
                            "host"
                        ],
                        "filter": "agent.type: filebeat"
                    },
                    {
                        "entity_types": [
                            "user"
                        ],
                        "filter": "user.name: ubuntu"
                    }
                ]
            },
            "references": [],
            "managed": false,
            "migrationVersion": {
                "risk-engine-configuration": "10.3.0"
            },
            "updated_at": "2025-09-29T06:52:46.483Z",
            "created_at": "2025-09-26T08:42:58.037Z",
            "version": "WzExLDFd",
            "coreMigrationVersion": "8.8.0",
            "typeMigrationVersion": "10.3.0",
            "score": 0
        }
    ]
}
```

### 2. **Preview Risk Scores with and without Filters**

I added 5 alerts for this test
host.name : "pessimistic-permafrost.name"
host.name : "yellowish-minority.info"
user.name : "Roscoe_Stehr-Murazik"
user.name : "Sheridan_MacGyver55"
user.name : "ubuntu"


**Without filters**
```bash
curl --location 'http://localhost:5601/internal/risk_score/preview' \
--header 'kbn-xsrf: true' \
--header 'elastic-api-version: 1' \
--header 'x-elastic-internal-origin: true' \
--header 'Content-Type: application/json' \
--header 'Authorization: Basic ****' \
--data '{
"data_view_id": ".alerts-security.alerts-default",
  "range": {
    "start": "now-30d",
    "end": "now"
  }}' | jq '{
  hosts: [.scores.host[] | {id_field, id_value}],
  users: [.scores.user[] | {id_field, id_value}]
}'

{
  "hosts": [
    {
      "id_field": "host.name",
      "id_value": "pessimistic-permafrost.name"
    },
    {
      "id_field": "host.name",
      "id_value": "yellowish-minority.info"
    }
  ],
  "users": [
    {
      "id_field": "user.name",
      "id_value": "Roscoe_Stehr-Murazik"
    },
    {
      "id_field": "user.name",
      "id_value": "Sheridan_MacGyver55"
    },
    {
      "id_field": "user.name",
      "id_value": "ubuntu"
    }
  ]
}
```

**With user filter**
```bash
curl --location 'http://localhost:5601/internal/risk_score/preview' \
--header 'kbn-xsrf: true' \
--header 'elastic-api-version: 1' \
--header 'x-elastic-internal-origin: true' \
--header 'Content-Type: application/json' \
--header 'Authorization: Basic ****' \
--data '{
"data_view_id": ".alerts-security.alerts-default",
  "range": {
    "start": "now-30d",
    "end": "now"
  },
  "filters": [
      {
        "entity_types": ["user"],
        "filter": "user.name: ubuntu"
      }
    ]
}' | jq '{
  hosts: [.scores.host[] | {id_field, id_value}],
  users: [.scores.user[] | {id_field, id_value}]
}'

{
  "hosts": [
    {
      "id_field": "host.name",
      "id_value": "pessimistic-permafrost.name"
    },
    {
      "id_field": "host.name",
      "id_value": "yellowish-minority.info"
    }
  ],
  "users": [
    {
      "id_field": "user.name",
      "id_value": "Roscoe_Stehr-Murazik"
    },
    {
      "id_field": "user.name",
      "id_value": "Sheridan_MacGyver55"
    }
  ]
}
```

**With user and host filter**
```bash
curl --location 'http://localhost:5601/internal/risk_score/preview' \
--header 'kbn-xsrf: true' \
--header 'elastic-api-version: 1' \
--header 'x-elastic-internal-origin: true' \
--header 'Content-Type: application/json' \
--header 'Authorization: Basic ****' \
--data '{
"data_view_id": ".alerts-security.alerts-default",
  "range": {
    "start": "now-30d",
    "end": "now"
  },
  "filters": [
      {
        "entity_types": ["user"],
        "filter": "user.name: ubuntu"
      },
      {
        "entity_types" : ["host"],
        "filter": "agent.type: filebeat"
      }
    ]
}' | jq '{
  hosts: [.scores.host[] | {id_field, id_value}],
  users: [.scores.user[] | {id_field, id_value}]
}'

{
  "hosts": [
    {
      "id_field": "host.name",
      "id_value": "pessimistic-permafrost.name"
    }
  ],
  "users": [
    {
      "id_field": "user.name",
      "id_value": "Roscoe_Stehr-Murazik"
    },
    {
      "id_field": "user.name",
      "id_value": "Sheridan_MacGyver55"
    }
  ]
}
```


### 3. **Test Invalid KQL Filter Handling**
```bash
curl -X POST "http://localhost:5601/api/risk_scores/preview" \
  -H "Content-Type: application/json" \
  -H "kbn-xsrf: true" \
  -d '{
    "data_view_id": "security-solution-default",
    "range": {
      "start": "now-30d",
      "end": "now"
    },
    "filters": [
      {
        "entity_types": ["host"],
        "filter": "invalid kql syntax {"
      }
    ]
  }'
```

**Expected Response:** Should return results without the invalid filter applied (graceful degradation)

## 🔄 Migration Strategy

- **Automatic Migration**: Existing saved objects are automatically migrated to include empty `filters` array

## 🎯 User Experience Impact

### Before
- Risk scoring applied to all alerts without filtering
- Limited control over which alerts contribute to risk scores
- No way to focus on specific environments or conditions

### After
- **Targeted Risk Scoring**: Filter alerts by environment, agent type, user groups, etc.
- **Flexible Configuration**: Apply different filters to different entity types

## 🔍 Key Technical Decisions

1. **Entity-Specific Filtering**: Filters are applied per entity type, allowing granular control
2. **KQL Integration**: Leverages existing Kibana Query Language for consistency
3. **Aggregation-Level Filtering**: Filters applied at ES aggregation level for performance

## 📝 Related Issues

- Closes: [elastic/security-team#13606](https://github.com/elastic/security-team/issues/13606)
- Related: [elastic/kibana#224229](https://github.com/elastic/kibana/issues/224229)

## 🚧 Next Steps

This PR implements the backend infrastructure. The frontend UI implementation will follow in a separate PR to:
- Add filter configuration UI components
- Integrate with existing risk engine configuration page
- Provide filter testing and validation features

---

**Note**: This is a backend-only implementation. Frontend changes will be delivered in a subsequent PR.

abhishekbhatia1710 committed:
feat: Add Risk Scoring Alert Filtering - Backend Implementation

- Add filters field to risk-engine-configuration saved object
- Implement migration logic for existing configurations
- Enhance risk scoring logic with entity-specific KQL filters
- Update API endpoints to support filter configuration and preview
- Add comprehensive unit tests for new functionality
- Maintain backward compatibility with existing configurations

Closes: elastic/security-team#13606

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

abhishekbhatia1710 committed:
Adding filters to the esql risk scoring

abhishekbhatia1710 committed:
Adding filters for the preview API

abhishekbhatia1710 committed:
Removing repeated code

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/check_mappings_update --fix'

kibanamachine committed:
[CI] Auto-commit changed files from 'yarn openapi:bundle'

kibanamachine committed:
[CI] Auto-commit changed files from 'make api-docs'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/jest_integration -u src/core/server/integration_tests/ci_checks'

abhishekbhatia1710 committed:
Implement entity-specific filtering for risk scoring

- Refactor both ES|QL and scripted metrics implementations to create separate queries for each entity type
- Apply custom filters only to specified entity types using entity_types field
- Fix TypeScript errors with proper type casting using Record<string, unknown> instead of any
- Ensure filters are applied as inclusion filters (must match) not exclusion filters

abhishekbhatia1710 committed:
Changing the filter query to filter out instead of filter in

abhishekbhatia1710 committed:
Fixing types

abhishekbhatia1710 committed:
Fixing tests :  updating mappings version

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/344733)
* Commit: 331594ad66550c2331a09b1c77ef815ca25c3d08

### Failed CI Steps
* [Defend Workflows Cypress Tests #18](https://buildkite.com/elastic/kibana-pull-request/builds/344733#0199944b-a81b-46f6-acb1-8db6e85c9a16)

### History
* :broken_heart: [Build #344203](https://buildkite.com/elastic/kibana-pull-request/builds/344203) failed ddac7b2eae590001a4184b035998618ada0a2fbc
* :broken_heart: [Build #343647](https://buildkite.com/elastic/kibana-pull-request/builds/343647) failed 46e2d8bffdd24ff8391518f75e836a24706f7016

cc @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/344733","number":344733,"commit":"331594ad66550c2331a09b1c77ef815ca25c3d08"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/344203","number":344203,"commit":"ddac7b2eae590001a4184b035998618ada0a2fbc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343647","number":343647,"commit":"46e2d8bffdd24ff8391518f75e836a24706f7016"}],"number":344733}
buildkite-pr-comment-->

jloleysens reviewed (approved):
Addition of optional `filters` field + mappings LGTM

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/344830)
* Commit: 5797e6e290f5504bf819400cab636296ed42c86f

### Failed CI Steps
* [Defend Workflows Cypress Tests #18](https://buildkite.com/elastic/kibana-pull-request/builds/344830#019994b1-a064-4d04-af6b-3cc580cb3ae5)

### History
* :broken_heart: [Build #344733](https://buildkite.com/elastic/kibana-pull-request/builds/344733) failed 331594ad66550c2331a09b1c77ef815ca25c3d08
* :broken_heart: [Build #344203](https://buildkite.com/elastic/kibana-pull-request/builds/344203) failed ddac7b2eae590001a4184b035998618ada0a2fbc
* :broken_heart: [Build #343647](https://buildkite.com/elastic/kibana-pull-request/builds/343647) failed 46e2d8bffdd24ff8391518f75e836a24706f7016

cc @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/344830","number":344830,"commit":"5797e6e290f5504bf819400cab636296ed42c86f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/344733","number":344733,"commit":"331594ad66550c2331a09b1c77ef815ca25c3d08"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/344203","number":344203,"commit":"ddac7b2eae590001a4184b035998618ada0a2fbc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343647","number":343647,"commit":"46e2d8bffdd24ff8391518f75e836a24706f7016"}],"number":344830}
buildkite-pr-comment-->

abhishekbhatia1710 committed:
Add filters to ReadRiskEngineSettingsResponse

- Settings route now returns filters array from saved object
- Updated schema to include filters in response
- Generated types updated to match schema

abhishekbhatia1710 committed:
Ensure filter exclusion logic in risk calculations

- Filters use must_not (exclusion) to filter out matching alerts
- Applied to both standard and ESQL calculation methods
- Aligns with customer use cases (filter out test data, service accounts, etc)

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

elasticmachine commented:
Pinging @elastic/obs-ux-management-team (Team:obs-ux-management)

CAWilson94 reviewed (approved):
Code Review only: LGTM

abhishekbhatia1710 committed:
Backend changes readded after resolving merge conflicts for the Risk score filtering

abhishekbhatia1710 committed:
Fixing types

abhishekbhatia1710 committed:
Merge branch 'main' into ea-13606-alert-filtering-risk-score-backend

abhishekbhatia1710 committed:
Snapshot updated manually as CI was failing while doing the same

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/354967)
* Commit: fe38fe01274a73c171911286cbe3d412f00a81f3

### Failed CI Steps
* [Check changes in Saved Objects](https://buildkite.com/elastic/kibana-pull-request/builds/354967#019a24c1-f66c-436a-96a6-e9d3a507086b)

### History
* :broken_heart: [Build #354935](https://buildkite.com/elastic/kibana-pull-request/builds/354935) failed 4f477046e047918edf9d3367435b0bc821161d9a
* :broken_heart: [Build #353166](https://buildkite.com/elastic/kibana-pull-request/builds/353166) failed e7bd7fccf91677ea985db295ba51117cc3ca75f8
* :broken_heart: [Build #351477](https://buildkite.com/elastic/kibana-pull-request/builds/351477) failed e7bd7fccf91677ea985db295ba51117cc3ca75f8
* :broken_heart: [Build #351454](https://buildkite.com/elastic/kibana-pull-request/builds/351454) failed 148a6d07e351c004709095978c0b3e094558ebe5
* :broken_heart: [Build #351311](https://buildkite.com/elastic/kibana-pull-request/builds/351311) failed 050ec1230cf31d691c07a3857ea8e0610301e0e6

cc @tiansivive @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354967","number":354967,"commit":"fe38fe01274a73c171911286cbe3d412f00a81f3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354935","number":354935,"commit":"4f477046e047918edf9d3367435b0bc821161d9a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353166","number":353166,"commit":"e7bd7fccf91677ea985db295ba51117cc3ca75f8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351477","number":351477,"commit":"e7bd7fccf91677ea985db295ba51117cc3ca75f8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351454","number":351454,"commit":"148a6d07e351c004709095978c0b3e094558ebe5"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351311","number":351311,"commit":"050ec1230cf31d691c07a3857ea8e0610301e0e6"}],"number":354967}
buildkite-pr-comment-->

abhishekbhatia1710 committed:
SO snapshot updates

abhishekbhatia1710 committed:
Fixing tests

abhishekbhatia1710 committed:
SO snapshot update...again

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/355587)
* Commit: 5e77a2d8aad225b2fd76946e791c6533f8a10bcd

### Failed CI Steps
* [Jest Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9d85-4816-ab71-6d22917ee6a7)
* [FTR Configs #29](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9dae-4711-b953-74c6a573c4e6)
* [FTR Configs #35](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9db5-4941-99e1-e056c168ed3f)
* [FTR Configs #39](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9db8-46cf-a8d5-f955fc24a037)
* [FTR Configs #54](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9dc6-4435-ba04-f96bf11c000d)
* [FTR Configs #56](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9dc8-4a74-839e-7dcee5034b85)
* [FTR Configs #59](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9dca-49d1-9152-813e0459f22d)
* [FTR Configs #70](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9dd3-4f13-99c7-4950f23abc31)
* [FTR Configs #72](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9dd6-4ee3-bb90-d03f1989ce27)
* [FTR Configs #72](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29f0-b18b-4755-9611-7ed081bd31ec)
* [FTR Configs #77](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9ddb-4364-a9d7-39818b2fd367)
* [FTR Configs #81](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9ddf-4287-acbb-e3af34545f9f)
* [FTR Configs #83](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9de0-4bf9-a9ed-51962f57b07c)
* [FTR Configs #98](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9ded-4f4f-863b-5c2b0791bd34)
* [FTR Configs #105](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9df5-4e99-84c7-1e47bbb47db5)
* [FTR Configs #108](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9df7-4d1c-b17d-38cae5e6ddb1)
* [FTR Configs #109](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9df8-41bc-8202-2b9d754a2a2c)
* [FTR Configs #111](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9df9-4819-9080-cb16bb042518)
* [FTR Configs #121](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9e01-4612-af1f-4b8658522aa1)
* [FTR Configs #125](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9e05-42f1-93d2-84abf08325d3)
* [FTR Configs #129](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9e08-4b4f-8bd2-335fa0a4bdd7)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9d85-4816-ab71-6d22917ee6a7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/355587/jobs/019a29d6-9d85-4816-ab71-6d22917ee6a7/artifacts/019a29f1-3e5b-4538-940b-6e8a5db8c2bc) Jest Tests #<span></span>8 / Calculate risk scores with ESQL ESQL query matches snapshot
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9db8-46cf-a8d5-f955fc24a037) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/355587/jobs/019a29d6-9db8-46cf-a8d5-f955fc24a037/artifacts/019a29fe-7de4-475c-9bc9-c092485cddc5) FTR Configs #<span></span>39 / Dataset Quality Dataset quality table shows the last activity
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9dca-49d1-9152-813e0459f22d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/355587/jobs/019a29d6-9dca-49d1-9152-813e0459f22d/artifacts/019a29fb-d3dd-43fb-80b6-5abc1096148c) FTR Configs #<span></span>59 / Entity Analytics - Risk Engine @<span></span>ess @<span></span>serverless @<span></span>serverlessQA init_and_status_apis init api should create configuration saved object
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9ddb-4364-a9d7-39818b2fd367) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/355587/jobs/019a29d6-9ddb-4364-a9d7-39818b2fd367/artifacts/019a29fb-2c33-4a34-b382-b3a6af8f0b89) FTR Configs #<span></span>77 / Entity Analytics - Risk Engine @<span></span>ess @<span></span>serverless @<span></span>serverlessQA init_and_status_apis init api should create configuration saved object
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9d85-4816-ab71-6d22917ee6a7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/355587/jobs/019a29d6-9d85-4816-ab71-6d22917ee6a7/artifacts/019a29f1-3e5b-4f80-951d-108702b1c962) Jest Tests #<span></span>8 / riskEngineConfigurationType version 3 migration should add filters field to existing configurations
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/355587#019a29d6-9d85-4816-ab71-6d22917ee6a7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/355587/jobs/019a29d6-9d85-4816-ab71-6d22917ee6a7/artifacts/019a29f1-3e5c-4055-909e-38500bc5de80) Jest Tests #<span></span>8 / riskEngineConfigurationType version 3 migration should preserve existing filters if they exist

### History
* :broken_heart: [Build #355582](https://buildkite.com/elastic/kibana-pull-request/builds/355582) failed 3179b691d40d5a8fc1b58f1a7e50ce7987419f42
* :broken_heart: [Build #354967](https://buildkite.com/elastic/kibana-pull-request/builds/354967) failed fe38fe01274a73c171911286cbe3d412f00a81f3
* :broken_heart: [Build #354935](https://buildkite.com/elastic/kibana-pull-request/builds/354935) failed 4f477046e047918edf9d3367435b0bc821161d9a
* :broken_heart: [Build #353166](https://buildkite.com/elastic/kibana-pull-request/builds/353166) failed e7bd7fccf91677ea985db295ba51117cc3ca75f8

cc @tiansivive @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355587","number":355587,"commit":"5e77a2d8aad225b2fd76946e791c6533f8a10bcd"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355582","number":355582,"commit":"3179b691d40d5a8fc1b58f1a7e50ce7987419f42"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354967","number":354967,"commit":"fe38fe01274a73c171911286cbe3d412f00a81f3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354935","number":354935,"commit":"4f477046e047918edf9d3367435b0bc821161d9a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353166","number":353166,"commit":"e7bd7fccf91677ea985db295ba51117cc3ca75f8"}],"number":355587}
buildkite-pr-comment-->

abhishekbhatia1710 committed:
Fixing tests

abhishekbhatia1710 committed:
Fixing types

abhishekbhatia1710 committed:
Fixing tests...part 2

abhishekbhatia1710 committed:
Merge branch 'main' into ea-13606-alert-filtering-risk-score-backend

abhishekbhatia1710 committed:
Fixing tests...part 3

abhishekbhatia1710 committed:
Merge branch 'ea-13606-alert-filtering-risk-score-backend' of github.com:abhishekbhatia1710/kibana into ea-13606-alert-filtering-risk-score-backend

abhishekbhatia1710 committed:
Fixing integration tests ... part 3

abhishekbhatia1710 committed:
Merge branch 'main' into ea-13606-alert-filtering-risk-score-backend
